sites/docs/src/content/docs/guidelines/pipelines/recommendations/testing.md

```diff
@@ -6,31 +6,155 @@ weight: 210
 
 ```tree
 tests
-├── single_end.main.nf.test
-└── paired_end.main.nf.test
+└── alignment-pair-end.nf.test
+└── alignment-single-end.nf.test
+├── default.nf.test
 ```
 
 ### Pipeline nf-tests additional guidance
 
-The same guidelines for test profiles, test data and nf-test also apply to pipeline tests. In addition, the following guidelines apply:
+The same guidelines for test profiles, test data and nf-test also apply to pipeline tests.
+In addition, the following guidelines apply:
 
 - To ensure all output files are caught, the `params.outdir` should be set the the nf-test variable `outputDir`
-- The tag `PIPELINE` and the pipeline name should be added to all tests
+- To ensure all output files are caught, it is highly recommended to use the `nft-utils` plugin to parse the output of the pipeline.
+- To ensure a proper execution of the pipeline is capture, it is recommended to capture the number of successful tasks.
+- To ensure that all output files are captured, it is recommended to run the tests once, and rerun to verify they're stable.
+  - If not such files can be added to the `.nftignore` file.
+
+```groovy
+nextflow_pipeline {
+
+    name "Test pipeline"
+    script "../main.nf"
+    tag "pipeline"
+
+    test("-profile test") {
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+        }
+
+        then {
+            // stable_name: All files + folders in ${params.outdir}/ with a stable name
+            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
+            // stable_path: All files in ${params.outdir}/ with stable content
+            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
+            assertAll(
+                { assert workflow.success},
+                { assert snapshot(
+                    // Number of successful tasks
+                    workflow.trace.succeeded().size(),
+                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
+                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_{{ short_name }}_software_mqc_versions.yml"),
+                    // All stable path name, with a relative path
+                    stable_name,
+                    // All files with stable contents
+                    stable_path
+                ).match() }
+            )
+        }
+    }
+}
+```
+
+`stable_name` capture all of the stable files and folders in the output directory.
+`stable_path` capture all of the stable files in the output directory, ignoring the files listed in the `.nftignore` file.
+
+Additional plugins can be used if files with unstable content can be read using a nf-test plugin.
+For example for BAM and CRAM files the `nft-bam` plugin is used, while the `nft-vcf` plugin is also used, both in conjonction with `nft-utils`:
```

Comment:
should we move this section into the nf-test assertion examples (and split it up into component tests and pipeline tests?